src/systems/media-frames.js

```diff
@@ -250,8 +250,13 @@ AFRAME.registerComponent("media-frame", {
 
   snapObject(capturedEl) {
     // TODO this assumes media frames are all in world space
-    capturedEl.object3D.position.copy(this.el.object3D.position);
-    capturedEl.object3D.rotation.copy(this.el.object3D.rotation);
+    const worldPosition = new THREE.Vector3();
```

Comment:
Not so bad since this isn't run every frame but we should avoid allocating these vectors every time, either by just pulling it out into the root of the file, or in a closure for this function alone. I will often just create a `const tmpWorldPos = new THREE.Vector3()` in a file to use for this sort of thing.

Reply:
Yeah, I thought that being non a tick function would be fine but it makes sense to avoid instancing if possible. Updated.